app/v1/hateoas/HateoasLinks.scala

```diff
@@ -28,13 +28,20 @@ trait HateoasLinks {
   private def metadataUri(appConfig: AppConfig, nino: String, calcId: String): String =
     baseUri(appConfig, nino) + s"/$calcId"
 
+  private def taxableIncomeUri(appConfig: AppConfig, nino: String, calcId: String): String =
+    baseUri(appConfig, nino) + s"/$calcId/taxable-income"
+
+
   // API resource links
-  def trigger(appConfig: AppConfig, nino: String) =
+  def trigger(appConfig: AppConfig, nino: String): Link =
     Link(href = baseUri(appConfig, nino), method = POST, rel = TRIGGER)
 
-  def list(appConfig: AppConfig, nino: String) =
+  def list(appConfig: AppConfig, nino: String): Link =
     Link(href = baseUri(appConfig, nino), method = GET, rel = SELF)
 
-  def getMetadata(appConfig: AppConfig, nino: String, calcId: String): Link =
-    Link(href = metadataUri(appConfig, nino, calcId), method = GET, rel = SELF)
+  def getMetadata(appConfig: AppConfig, nino: String, calcId: String, isSelf: Boolean): Link =
```

Comment:
```suggestion
  def getMetadata(appConfig: AppConfig, nino: String, calcId: String, isSelf: Boolean = True): Link =
```

If we default the value of the boolean here then we don't need to change the code for the existing endpoints which are already merged in.